File: src/MetaDSL/test/models/ts/Facture.ts
import { Mission } from './Mission';

export class Facture {
    public readonly id: string;  // ID en lecture seule

    public readonly invoiceNumber: string;  // Le numéro de facture est visible uniquement si la mission est du type "facturé"

    public mission: Mission;  // Relation avec la mission

    constructor(id: string, invoiceNumber: string, mission: Mission) {
        this.id = id;
        this.invoiceNumber = invoiceNumber;
        this.mission = mission;
    }
}

File: src/MetaDSL/test/models/dotnet/Facture.cs
using System;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

public class Facture
{
    [Key]
    public Guid Id { get; private set; }

    [JsonIgnore]  // Le numéro de facture est visible uniquement si la mission est du type "facturé"
    public string InvoiceNumber { get; private set; }

    [ForeignKey("MissionId")]
    public virtual Mission Mission { get; set; }
    public Guid MissionId { get; set; }

    public Facture(string invoiceNumber)
    {
        Id = Guid.NewGuid();
        InvoiceNumber = invoiceNumber;
    }
}
